modulemd/v2/include/modulemd-2.0/private/modulemd-yaml.h

```diff
@@ -421,4 +421,129 @@ modulemd_yaml_emit_document_headers (yaml_emitter_t *emitter,
                                      guint64 mdversion,
                                      GError **error);
 
+
+/* A set of macros for simple emitting of common elements */
+#define NON_EMPTY_TABLE(table) g_hash_table_size (table) != 0
+
+#define EMIT_KEY(emitter, error, value)                                       \
+  do                                                                          \
+    {                                                                         \
+      if (!mmd_emitter_scalar (                                               \
+            emitter, value, YAML_PLAIN_SCALAR_STYLE, error))                  \
+        return FALSE;                                                         \
+    }                                                                         \
+  while (0)
+
+#define EMIT_SCALAR_IF_SET(emitter, error, key, value)                        \
+  do                                                                          \
+    {                                                                         \
+      if (value != NULL)                                                      \
+        {                                                                     \
+          EMIT_KEY (emitter, error, key);                                     \
+          if (!mmd_emitter_scalar (                                           \
```

Comment:
If we rename `EMIT_KEY()` to `EMIT_SCALAR()`, you can just do both here and it'll be shorter and more easily read.